internal/pkg/wirecli/wirek8s/wirek8s.go

```diff
@@ -73,36 +90,64 @@ func NewRestConfig(master clik8s.MasterURL, path clik8s.KubeConfigPath) (*rest.C
 	return clientcmd.BuildConfigFromFlags(string(master), string(path))
 }
 
+// NewPluginConfig returns a new PluginConfig
+func NewPluginConfig() *types.PluginConfig {
+	pc := plugin.DefaultPluginConfig()
+	pc.GoEnabled = true
+	return pc
+}
+
+// NewResMapFactory returns a rew ResMap Factory
+func NewResMapFactory(pc *types.PluginConfig) *resmap.Factory {
+	uf := kunstruct.NewKunstructuredFactoryWithGeneratorArgs(
+		&types.GeneratorMetaArgs{
+			PluginConfig: pc,
+		})
+	return resmap.NewFactory(resource.NewFactory(uf))
+}
+
+// NewTransformerFactory returns a new transformer factory
+func NewTransformerFactory() transformer.Factory {
+	return ktransformer.NewFactoryImpl()
+}
+
+// NewFileSystem returns a new filesystem
+func NewFileSystem() fs.FileSystem {
+	return fs.MakeRealFS()
+}
+
+// NewConfigProvider returns a new ConfigProvider
+func NewConfigProvider(rf *resmap.Factory, fSys fs.FileSystem, tf transformer.Factory, pc *types.PluginConfig) resourceconfig.ConfigProvider {
+	return &resourceconfig.KustomizeProvider{
+		RF: rf,
+		TF: tf,
+		FS: fSys,
+		PC: pc,
+	}
+}
+
 // NewKubernetesClientSet provides a clientset for talking to k8s clusters
 func NewKubernetesClientSet(c *rest.Config) (*kubernetes.Clientset, error) {
 	return kubernetes.NewForConfig(c)
 }
 
-// NewFileSystem provides a real FileSystem
-func NewFileSystem() fs.FileSystem {
-	return fs.MakeRealFS()
+// NewDynamicClient provides a dynamic client
+func NewDynamicClient(c *rest.Config) (dynamic.Interface, error) {
```

Comment:
Why a dynamic and a controller runtime client ?
Can we use dynamic client directly ?

Reply:
I'll use dynamic client instead of controller runtime client.